controllers/predicate/predicate.go

```diff
@@ -205,3 +189,33 @@ func IsSnapshotLease() predicate.Predicate {
 		},
 	}
 }
+
+// JobStatusChanged is a predicate that is `true` if the status of a job changes.
+func JobStatusChanged() predicate.Predicate {
```

Comment:
Why is this predicate required?

Reply:
To trigger the reconciler when a job status is changed

Reply:
When a Job is created then a re-queue is already done by the compaction reconciler. There is only one case which is etcd-druid restarts after spawning a Job. In that case also it will still get an event for the Job. Previously this predicate was not there, therefore my question on why is it required now

Reply:
Mainly to update the job related metrics correctly

Reply:
We discussed this in the meeting on why do we need it